docs/developer_guide/framework_details.rst

```diff
@@ -1,19 +1,23 @@
 Framework Details
 =================
 
-The Lewis framework is built around a cycle-based statemachine that
-drives the device simulation, and shared protocol adapters that separate
-the communication layer from the simulated device.
+The Lewis framework is built around a cycle-driven core which drives the
```

Comment:
I noticed this as well, but couldn't really think of better options. I like the "in turn" version. 

The other option would be to go for something more general like "cycle-driven core which manages ...". Or leave it as "cycle-based core which drives"... but that creates inconsistency with based vs driven. Maybe I should just switch them all back to "cycle-based"?